pkg/config/staging.go

```diff
@@ -113,6 +113,43 @@ func (bp *Blueprint) evalGhpcStageInModuleSettings() error {
 	return errs.OrNil()
 }
 
+// Evaluate a terraform provider as an intermediate step
+func evalGhpcStageInProvider(providers map[string]TerraformProvider, path mapPath[providerPath], ctx *hcl.EvalContext) Errors {
+	errs := Errors{}
+	for k1, v1 := range providers {
+		us := map[string]cty.Value{}
+		for k2, v2 := range v1.Configuration.Items() {
```

Comment:
nit. What I meant is to reduce bolierplate code by unifying logic for providers and modules usinf `evalGhpcStageInDict`

Reply:
Ah ok, complete.